### PR TITLE
Save and restore terminal settings around commands Fixes #69

### LINKED ISSUE #69: After abruptly terminating hammer-vlsi, the terminal needs reset
The terminal is left in a weird state and I have to clear or reset it to normalize.

### PR DESCRIPTION
**Pull request by colinschmidt**
Tested that this works for me when I ctrl+c during a tool run

### CODE REVIEW

**Comment by colinschmidt** [comment]
FYI: I followed something similar to this example: https://docs.python.org/3/library/termios.html#example

**Review by edwardcwang — APPROVED** [review]

**Review by harrisonliew — APPROVED** [review]
Seems like the CI is unhappy about termios, but otherwise, LGTM.

File: src/hammer-vlsi/hammer_vlsi/submit_command.py [hunk 1]
```diff
@@ -123,12 +125,18 @@ def submit(self, args: List[str], env: Dict[str, str],
         # Just run the command on this host.
 
         prog_tag = self.get_program_tag(args)
+        term_settings = termios.tcgetattr(sys.stdin.fileno())
```
Comment (edwardcwang) [inline src/hammer-vlsi/hammer_vlsi/submit_command.py:128]:
```suggestion
        term_settings: Optional[list] = None
        try:
            termios.tcgetattr(sys.stdin.fileno())
        except termios.error:
            # Command should still run even if some tty features are not present
            pass
```

File: src/hammer-vlsi/hammer_vlsi/submit_command.py [hunk 2]
```diff
@@ -123,12 +125,18 @@ def submit(self, args: List[str], env: Dict[str, str],
...
 
         logger.debug("Executing subprocess: " + ' '.join(args))
         subprocess_logger = logger.context("Exec " + prog_tag)
         proc = subprocess.Popen(args, shell=False, stderr=subprocess.STDOUT,
                                 stdout=subprocess.PIPE, env=env, cwd=cwd)
+        # These are run in reverse order of registration
+        # Terminate is first to allow for the possibility of graceful shutdown
+        # Then the program will be forceably terminated and we restore the terminal settings
+        atexit.register(termios.tcsetattr, sys.stdin.fileno(), termios.TCSANOW, term_settings)
```
Comment (edwardcwang) [inline src/hammer-vlsi/hammer_vlsi/submit_command.py:137]:
```suggestion
        if term_settings is not None:
            atexit.register(termios.tcsetattr, sys.stdin.fileno(), termios.TCSANOW, term_settings)
```

**Review by edwardcwang — CHANGES_REQUESTED** [review]
Hmm how about something like this?

**Comment by edwardcwang** [comment]
(didn't actually test but something like this idea)

**Comment by colinschmidt** [comment]
I did a slightly more "intent-full" design. We only need to worry about terminal settings if we are a tty.

**Review by edwardcwang — COMMENTED** [review]
Will squash merge once the PR updates

**Review by edwardcwang — APPROVED** [review]

**Pull request merged**